src/main/java/com/redhat/devtools/intellij/quarkus/module/QuarkusExtensionsStep.java

```diff
@@ -236,24 +167,41 @@ protected void customizeCellRenderer(@NotNull JList<? extends QuarkusExtension>
             selectedExtensionsPanel.add(label);
             selectedExtensionsPanel.add(selectedExtensions);
-            panel.setSecondComponent(new JBScrollPane(selectedExtensionsPanel));
-            extensionsTable.getModel().addTableModelListener(new TableModelListener() {
+            extensionsPanel.setSecondComponent(new JBScrollPane(selectedExtensionsPanel));
+            panel.add(extensionsPanel);
+
+            filter.addDocumentListener(new DocumentAdapter() {
+                @Override
+                protected void textChanged(@NotNull DocumentEvent e) {
+                    ApplicationManager.getApplication().invokeLater(() -> {
+                        extensionsTree.setModel(new DefaultTreeModel(getModel(categories, filter)));
+                        expandTree(extensionsTree);
+                    });
+                }
+            });
+            extensionDetailTextPane.addHyperlinkListener(new HyperlinkListener() {
                 @Override
-                public void tableChanged(TableModelEvent e) {
-                    if (e.getType() == TableModelEvent.UPDATE) {
-                        selectedExtensions.setModel(new SelectedExtensionsModel(categories));
+                public void hyperlinkUpdate(HyperlinkEvent e) {
+                    if (e.getEventType() == HyperlinkEvent.EventType.ACTIVATED) {
+                        BrowserUtil.browse(e.getURL());
                     }
                 }
             });
-            extensionsTable.getSelectionModel().addListSelectionListener(new ListSelectionListener() {
+            extensionsTree.addCheckboxTreeListener(new CheckboxTreeListener() {
+                @Override
+                public void nodeStateChanged(@NotNull CheckedTreeNode node) {
+                    ((QuarkusExtension) node.getUserObject()).setSelected(node.isChecked());
+                    selectedExtensions.setModel(new SelectedExtensionsModel(categories));
+                }
+            });
+            extensionsTree.getSelectionModel().addTreeSelectionListener(new TreeSelectionListener() {
                 @Override
-                public void valueChanged(ListSelectionEvent e) {
-                    int index = extensionsTable.getSelectedRow();
-                    if (index >=0) {
-                        QuarkusExtension extension = ((ExtensionsTable.Model)extensionsTable.getModel()).getElementAt(index);
-                        StringBuilder builder = new StringBuilder("<html><body>" + extension.getDescription() + ".");
-                        if (StringUtils.isNotBlank(extension.getGuide())) {
-                            builder.append(" <a href=\"" + extension.getGuide() + "\">Click to open guide</a>");
+                public void valueChanged(TreeSelectionEvent e) {
+                    Object comp = ((DefaultMutableTreeNode) e.getNewLeadSelectionPath().getLastPathComponent()).getUserObject();
```

Comment:
Still face the NPE but now i know when it happens

If you select an option and then type a word that does not return the selected option (E.g select something then type `zz`) , `e.getNewLeadSelectionPath()` will return null causing the NPE